src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpCookieHelper.cs

```diff
@@ -0,0 +1,95 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+
+using SafeWinHttpHandle = Interop.WinHttp.SafeWinHttpHandle;
+
+namespace System.Net.Http
+{
+    internal static class WinHttpCookieHelper
```

Comment:
I'll change the name of the class to `WinHttpCookieContainerAdapter`.  Longer term, I plan to refactor this code and use a nested class, separate file pattern, similar to CurlHandler.  I.e., make these "helper" classes subclasses of WinHttpHandler but split into separate source files, i.e. WinHttpHandler.CookieContainerAdapter.cs, etc.  This would also be more consist with the project coding guidelines.

However, I don't want to do a "new" of this class.  The `state` object is created for each request.  But the logic of the "adapter" is fixed.  So, it actually makes sense to pass in a `state` object to the methods.

I don't want to block on this for this PR.  But if there are other ideas, I can consider for next time.